gnosis/verification.k

```diff
@@ -72,22 +72,52 @@ module VERIFICATION
 
 // Reading symbolic WordStack from memory, then extracting bytes from it. General case (stack size can be != 32)
 //========================================================================
-    rule nthbyteof(chop(#asWord(W : WS) &Int MASK), I       , N       )
-      => nthbyteof(chop(#asWord(    WS) &Int MASK), I -Int 1, N -Int 1)
-      requires I >Int 0
+    //decoding nthbyteof(#asWord())
+    rule nthbyteof(#asWord(W : WS), I       , N       )
+      => nthbyteof(#asWord(    WS), I -Int 1, N -Int 1)
+      requires I >Int 0 andBool #sizeWordStack(W : WS) <=Int 32
```

Comment:
`#sizeWordStack(W : WS) <=Int N`

Reply:
Here it's correct. `#sizeWordStack(W : WS) <=Int 32` is required to force #asWord trimming rule to apply.

Reply:
Counter example.

```
nthbyteof(#asWord(W1 : W2 : W3 : .WordStack), 1, 4)  ==  W1
nthbyteof(#asWord(     W2 : W3 : .WordStack), 0, 3)  ==  0
```


Also, not sure it is OK without requiring `I <Int N`.

Reply:
My suggestion:

`0 <Int I andBool I <Int N andBool N <=Int 32` and

```
andBool N <=Int #sizeWordStack(W : WS)
```
or
```
andBool N ==Int #sizeWordStack(W : WS)
```

Reply:
Oh, now I see the issue with this lemma. Corrected.